### ISSUE TITLE
aria-label for native button element inside ic-button does not get updated if value changes

### ISSUE THREAD

**Issue by ad9242** [mi6/ic-ui-kit#1110]
## Summary of the bug 
If you set an aria-label on an ic-button and then subsequently change the value, the native button element inside does not get update with the new value, causing screen readers to read the wrong value.

## 🪜 How to reproduce 
Tell us the steps to reproduce the problem: 
1. set the aria-label attribute on an ic-button
2. later change the attribute to another value 
3. inspect the <button> element in browser dev tool
4. the aria label is still set to the original value

## 🖥 📱 Device 
All

## 🧐 Expected behaviour 
The aria-label should be updated

## 📝 Acceptance Criteria  
If relevant, describe in full detail the different interactions and edge cases that the component or patterns needs to fulfil. 

*Given* an aria label is updated on an ic-button component
*When*  a user focus the button with a screen reader enabled
*Then* the updated label should be read by the screen reader

## Additional info 
For aria-describedby, a mutation observer was added to watch when the value changes. A similar approach may be possible to resolve this

This is needed for DatePicker component.

**Comment by ad9242** [comment]
could be easier to fix after resolving #1131
